### PR TITLE
Fix sentinel refresh and nil pointer issue

### PR DESCRIPTION
**Pull request by linglingye001**

### CODE REVIEW

File: internal/controller/processor.go [hunk 1]
```diff
@@ -76,6 +76,7 @@ func (processor *AppConfigurationProviderProcessor) processFullReconciliation()
 	processor.RefreshOptions.ConfigMapSettingPopulated = true
 	processor.RefreshOptions.updatedKeyValueETags = updatedSettings.KeyValueETags
 	processor.RefreshOptions.updatedFeatureFlagETags = updatedSettings.FeatureFlagETags
+	processor.RefreshOptions.updatedSentinelETags = updatedSettings.SentinelETags
```
Comment (RichardChen820) [inline internal/controller/processor.go:79]: Shall we check if the returned `updatedSettings.SentinelETags` is not nil, then we also change the `processor.RefreshOptions.sentinelChanged=true`? Then we don't have to add additional check in `Finish()`

Comment (RichardChen820) [reply internal/controller/processor.go:79]: And we can also ensure that the `processor.ReconciliationState.SentinelETags` would never be set to nil

File: internal/loader/configuration_setting_loader.go [hunk 2]
```diff
@@ -299,22 +315,14 @@ func (csl *ConfigurationSettingLoader) CheckAndRefreshSentinels(
...
-	}
 	refreshedETags := make(map[acpv1.Sentinel]*azcore.ETag)
-	sentinels := normalizeSentinels(provider.Spec.Configuration.Refresh.Monitoring.Sentinels)
-
-	for _, sentinel := range sentinels {
-		if eTag, ok := eTags[sentinel]; ok {
-			// Initialize the updatedETags with the current eTags
-			refreshedETags[sentinel] = eTag
-		}
+	for sentinel, currentETag := range eTags {
```
Comment (RichardChen820) [inline internal/loader/configuration_setting_loader.go:319]: Do we need to add a nil check for the `eTags`?

File: internal/controller/processor.go (hunk 1 above)
Comment (linglingye001) [reply internal/controller/processor.go:79]: There may be 2 scenarios when shouldReconcile is true. User adds monitoring section or removes it. If no sentinel is setup, we need to make sure the `ReconciliationState.SentinelETags` be updated to empty list.

Comment (linglingye001) [reply internal/controller/processor.go:79]: I agree that sentinelETags should not be nil.

**Review by RichardChen820 — APPROVED** [review]

**Pull request merged**